### ISSUE TITLE
[Enhancement]: Data Source: aws_identitystore_group list all groups

### ISSUE THREAD

**Issue by wesleung** [hashicorp/terraform-provider-aws#31807]
### Description

Would like to get a listing of all identitystore groups for a specific identitystore
Like this AWS cli command:
`aws identitystore list-groups --identity-store-id d-XXXXXXXXXXX--query Groups[].GroupId`

Currently Data Source: aws_identitystore_group requires
- identity_store_id
- alternate_identifier or group_id 
The requirement to have "alternate_identifier or group_id" prevents a listing of all groups. Would like to make this optional so all groups would be returned for a specific "identity_store_id"

### Affected Resource(s) and/or Data Source(s)

- aws_identitystore_group

### Potential Terraform Configuration

```terraform
data "aws_ssoadmin_instances" "current" {}
data "aws_identitystore_group" "current" {
  identity_store_id = one(data.aws_ssoadmin_instances.current.identity_store_ids)
}
```

### Would you like to implement a fix?

None

**Comment by bigjimmynz** [comment]
It would be inline with this AWS API and [CLI method](https://docs.aws.amazon.com/cli/latest/reference/identitystore/list-groups.html)

**Comment by crablab** [comment]
There's a more general point here, in that the Provider doesn't support querying the full range of endpoints that are available. For example: 
- [list-group-memberships](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/identitystore/list-group-memberships.html) - shows the UIDs that belong to a group
- [list-groups](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/identitystore/list-groups.html)
- [list-users](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/identitystore/list-users.html)

`list-groups` and `list-group-memberships` are particularly useful as they enable you to know which SSO users are in a given SSO group. There's a Resource for [aws_identitystore_group_membership](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/identitystore_group_membership) but no equivalent way to retrieve that via Data. 

The (equivalent in IAM](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/iam_group)] doesn't support listing group, but does return group members and allow listing of users. 

As a use case, we use AWS SSO across our accounts with users and groups SCIM synced in from our SSO provider. In a sub-account we need to deploy an application which doesn't support SCIM sync but does support SAML. We would like to use Terraform to retrieve the users/groups and map them into the application so we can avoid manually reconfiguring users/groups in that app, accepting that it's not ideal users won't automatically be provisioned.

**Comment by cmanzur** [comment]
Hello, I tried for hours until I reach the following solution. I hope it helps you! :)

This will output all the groups:

```
data "external" "aws_groups" {
  program = ["sh", "-c", "RESULT=$(aws identitystore list-groups --identity-store-id ${local.identity_store_id} | jq '.Groups' | base64 -w0); jq -n --arg RESULT $RESULT '{\"base64\": $RESULT}'"]
}

output "aws_groups" {
  value = base64decode(data.external.aws_groups.result.base64)
}
```

**Comment by aristosvo** [comment]
I believe the original enhancement request is fulfilled by #36993 (implementing data source `aws_identitystore_groups`) and therefore could be closed.

**Comment by bschaatsbergen** [comment]
Closing, `data.aws_identitystore_group` has been released under v5.47.0 (#36993)
